src/tools/compiletest/src/runtest.rs

```diff
@@ -350,10 +350,13 @@ impl<'test> TestCx<'test> {
             }
         } else {
             if proc_res.status.success() {
-                self.fatal_proc_rec(
-                    &format!("{} test compiled successfully!", self.config.mode)[..],
-                    proc_res,
-                );
+                {
+                    self.error(&format!("{} test did not emit an error", self.config.mode));
```

Comment:
Remark (for myself): improve this diagnostics when I get to untangling the runtest logic to make it so that test suites explicitly declare their intended default behavior so that the diagnostics here can say e.g. "expected to check-fail, but this didn't fail" or whatever.